spec/models/partner_spec.rb

```diff
@@ -151,11 +151,57 @@
     end
   end
 
-  describe '#invite_new_partner' do
+  describe 'changing emails' do
     let(:partner) { create(:partner) }
 
-    it "should call the PartnerUser.invite! when the partner is changed" do
+    it "should not call the PartnerUser.invite! when the partner has status uninvited and the email is changed" do
```

Comment:
Since this is a pretty repeated, you can use a loop to write all the specs:
```
[:uninvited, :deactivated].each do |...|
   ...
end
```
This is optional though!